test_refreshable_mv is flaky

### test_refreshable_mv_in_replicated_db

```
        # Drop all tables and check that coordination znodes were deleted.
        for name, uuid, coordinated in tables:
            sync = randint(0, 1) == 0
            nodes[randint(0, 1)].query(f"drop table re.{name}{' sync' if sync else ''}")
            # TODO: After https://github.com/ClickHouse/ClickHouse/issues/61065 is done (for MVs, not ReplicatedMergeTree), check the parent znode instead.
            if sync:
>               assert not znode_exists(uuid)
E               AssertionError: assert not True
E                +  where True = <function test_refreshable_mv_in_replicated_db.<locals>.<lambda> at 0x7fe60b23bf40>('24c70c9d-561c-47d8-a419-e06ddace7bc1')
```

### test_replicated_db_startup_race

Fails due to previous test leaved the `re` database

```
E           helpers.client.QueryRuntimeException: Client failed! Return code: 82, stderr: Received exception from server (version 25.5.1):
E           Code: 82. DB::Exception: Received from 172.16.4.5:9000. DB::Exception: Database re already exists.. Stack trace:
```

See also:
- https://github.com/ClickHouse/ClickHouse/issues/76418

Cc: @al13n321

--- a/tests/integration/test_refreshable_mv/test.py
+++ b/tests/integration/test_refreshable_mv/test.py
@@ -4,7 +4,7 @@ from random import randint
 
 import pytest
 
-from helpers.cluster import ClickHouseCluster
+from helpers.cluster import ClickHouseCluster, QueryRuntimeException
 from helpers.network import PartitionManager
 from helpers.test_tools import assert_eq_with_retry, assert_logs_contain
 
@@ -354,8 +354,13 @@ def test_pause(started_cluster, cleanup):
     assert node2.query("select * from re.a") == "1\n"
     node2.query("system stop replicated view re.a")
     node1.restart_clickhouse() # just to guarantee that it notices the new znode
+    try:
+        node2.query("system wait view re.a")
+    except QueryRuntimeException as ex:
+        # If the node1.restart_clickhouse() interrupted a refresh, the error message (with substring
+        # "cancelled") is written to keeper, then thrown by "system wait view". That's normal.
+        assert "cancelled" in str(ex)
     node2.query(
-        "system wait view re.a;"
         "truncate table re.src;"
         "insert into re.src values (2);")
     time.sleep(3)
@@ -415,6 +420,8 @@ def do_test_backup(to_table):
 
     assert node1.query(tables_exist_query) == "2\n"
     assert node2.query(tables_exist_query) == "2\n"
+    node1.query(f'SYSTEM SYNC REPLICA re.{target}')
+    node2.query(f'SYSTEM SYNC REPLICA re.{target}')
     assert node1.query(f'SELECT * FROM re.{target}') == '1\n'
     assert node2.query(f'SELECT * FROM re.{target}') == '1\n'
 

